R/as_reporting_triangle.R

```diff
@@ -0,0 +1,212 @@
+#' @title Create a reporting triangle
+#'
+#' @param data Either a matrix of a reporting triangle or a
+#'    data.frame to be converted to a reporting triangle matrix.
+#'    Can either be in the long tidy format of counts by reference date
+#'    and report date or line list data with individual observations
+#'    indexed by their reference date and report date.
+#' @param max_delay Integer indicating the maximum delay to estimate.
+#' @param ... Additional arguments passed to methods. For data.frame method:
+#'    `strata`, `reference_date_col_name`, `report_date_col_name`,
+#'    `count_col_name`, `delays_unit`. For matrix method: `reference_dates`,
+#'    `strata`, `delays_unit`.
+#
+#' @returns `reporting_triangle` class object which is a list containing:
+#'    - A matrix with which rows are reference times and columns are delays and
+#'    entries are incident cases at each reference time and delay.
+#'    - An integer indicating the maximum delay used to create the reporting
+#'    triangle
+#'    - A vector of the same length as the rows of the matrix indicating the
+#'    dates corresponding to the reference times in the rows of the reporting
+#'    triangle.
+#'    - A character string indicating the strata.
+#'    - A vector indicating the "structure" of the reporting triangle.
+#'    - A character string indicating the unit of the delays.
+#' @export
+#'
+#' @examples
+#' data_as_of_df <- syn_nssp_df[syn_nssp_df$report_date <= "2026-04-01", ]
+#' as_reporting_triangle(
+#'   data = data_as_of_df,
+#'   max_delay = 25
+#' )
+#' @importFrom lubridate time_length days ymd
+as_reporting_triangle <- function(data, max_delay, ...) {
+  UseMethod("as_reporting_triangle")
+}
+
+#' @export
+as_reporting_triangle.default <- function(data, max_delay, ...) {
```

Comment:
you really don't want or need to do this imo. I would also suggest just making the default method be a matrix

Reply:
have removed